src/Workspaces/Core/Portable/Shared/Extensions/ICompilationExtensions.cs

```diff
@@ -133,5 +135,32 @@ public static INamedTypeSymbol SuppressMessageAttributeType(this Compilation com
         {
             return compilation.GetTypeByMetadataName("System.Diagnostics.CodeAnalysis.SuppressMessageAttribute");
         }
+
+        public static bool TryReduceExtensionMethod(
+            this Compilation compilation, IMethodSymbol method, ITypeSymbol receiver, CancellationToken cancellationToken,
+            out IMethodSymbol reducedMethod)
+        {
+            // Map 'rethod' and 'receiver' to the same language.  That way we can actually call
+            // 'ReduceExtensionMethod' below.  Symbol methods require all arguments to be from
+            // the same language as the symbol itself.
+            if (method != null && method.Language != compilation.Language)
+            {
+                method = SymbolFinder.FindSimilarSymbols(method, compilation, cancellationToken).FirstOrDefault();
+            }
+
+            if (receiver != null && receiver.Language != compilation.Language)
+            {
+                receiver = SymbolFinder.FindSimilarSymbols(receiver, compilation, cancellationToken).FirstOrDefault();
+            }
+
+            if (method == null || receiver == null)
+            {
+                reducedMethod = null;
+                return false;
+            }
+
+            reducedMethod = method.ReduceExtensionMethod(receiver);
```

Comment:
are we guaranteed that FindSimilarSymbols always returns a symbol that matches the compilation language or null?